app/service/operation_svc.py

```diff
@@ -106,3 +129,27 @@ def __init__(self):
         planning_module = import_module(chosen_planner[0]['module'])
         return getattr(planning_module, 'LogicalPlanner')(operation, self.get_service('planning_svc'),
                                                           **chosen_planner[0]['params'])
+
+    @staticmethod
+    async def _get_all_possible_abilities_by_agent(hosts, adversary):
+        return {a['paw']: {'agent_id': a['id'], 'all_abilities': [ab for p in adversary['phases']
+                                                                  for ab in adversary['phases'][p]]} for a in hosts}
+
+    async def _check_reason_skipped(self, agent, ability, op_facts, op_results, state, agent_executors):
+        variables = re.findall(r'#{(.*?)}',
+                               await self.get_service('planning_svc').decode(ability['test'], agent,
+                                                                             agent['host_group']),
+                               flags=re.DOTALL)
+        if ability['platform'] != agent['platform']:
```

Comment:
I like reason_id - but I think instead you should use an Enum like we're doing with LinkState (see base_service)